File: elixir/p2.exs
fib = fn
  a,b,0 -> a
  a,b,n -> fib.(b, a+b, n-1)
end
IO.puts Enum.sum(Enum.filter(Enum.map(1..50, fn n -> fib.(0,1,n) end ), even and fn(x) -> x < 4000000 end))

File: python/p2.py
def fib(a,b,n):
  if n == 0:
    return a
  return fib(b, a + b, n - 1)

i = 0
s = 0
f = 0 
while f < 4000000:
  f = fib(0, 1, i)
  if f % 2 == 0:
    s += f
  i += 1

print(s)